homeassistant/components/mqtt/light/schema_basic.py

```diff
@@ -298,47 +315,44 @@ def brightness_received(msg):
             self._brightness = percent_bright * 255
             self.async_write_ha_state()
 
-        if self._topic[CONF_BRIGHTNESS_STATE_TOPIC] is not None:
-            topics[CONF_BRIGHTNESS_STATE_TOPIC] = {
-                "topic": self._topic[CONF_BRIGHTNESS_STATE_TOPIC],
-                "msg_callback": brightness_received,
-                "qos": self._config[CONF_QOS],
-            }
-        elif (
-            self._optimistic_brightness
-            and last_state
-            and last_state.attributes.get(ATTR_BRIGHTNESS)
-        ):
-            self._brightness = last_state.attributes.get(ATTR_BRIGHTNESS)
+        add_topic(CONF_BRIGHTNESS_STATE_TOPIC, brightness_received)
+        restore_state(ATTR_BRIGHTNESS, self._optimistic_brightness)
```

Comment:
In previous implementation the `restore_state` would not run if `CONF_BRIGHTNESS_STATE_TOPIC` is not None, now it will run anyhow? is it important to keep the same behavior? if so we can return a flag from `add_topic` and change code to something like this:
```python
if not add_topic(CONF_BRIGHTNESS_STATE_TOPIC, brightness_received):
    restore_state(ATTR_BRIGHTNESS, self._optimistic_brightness)
```

Reply:
It should be more correct to only check `self._optimistic_brightness`, and if that flag is set we restore the state.